docs/user-guide/node-selection/node-selection.md

```diff
@@ -0,0 +1,200 @@
+
+### Step One: Attach label to the node
+
+Run `kubectl get nodes` to get the names of your cluster's nodes. Pick out the one that you want to add a label to.
+
+Then, to add a label to the node you've chosen, run `kubectl label nodes <node-name> <label-key>=<label-value>`. For example, if my node name is 'kubernetes-foo-node-1.c.a-robinson.internal' and my desired label is 'disktype=ssd', then I can run `kubectl label nodes kubernetes-foo-node-1.c.a-robinson.internal disktype=ssd`.
+
+If this fails with an "invalid command" error, you're likely using an older version of kubectl that doesn't have the `label` command. In that case, see the [previous version](https://github.com/kubernetes/kubernetes/blob/a053dbc313572ed60d89dae9821ecab8bfd676dc/examples/node-selection/README.md) of this guide for instructions on how to manually set labels on a node.
+
+Also, note that label keys must be in the form of DNS labels (as described in the [identifiers doc](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/design/identifiers.md)), meaning that they are not allowed to contain any upper-case letters.
+
+You can verify that it worked by re-running `kubectl get nodes --show-labels` and checking that the node now has a label.
+
+### Step Two: Add a nodeSelector field to your pod configuration
+
+Take whatever pod config file you want to run, and add a nodeSelector section to it, like this. For example, if this is my pod config:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    env: test
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+```
+
+Then add a nodeSelector like so:
+
+{% include code.html language="yaml" file="pod.yaml" ghlink="/docs/user-guide/node-selection/pod.yaml" %}
+
+When you then run `kubectl create -f pod.yaml`, the pod will get scheduled on the node that you attached the label to! You can verify that it worked by running `kubectl get pods -o wide` and looking at the "NODE" that the pod was assigned to.
+
+## Interlude: built-in node labels
+
+In addition to labels you [attach yourself](#step-one-attach-label-to-the-node), nodes come pre-populated
+with a standard set of labels. As of Kubernetes v1.4 these labels are
+
+* `kubernetes.io/hostname`
+* `failure-domain.beta.kubernetes.io/zone`
+* `failure-domain.beta.kubernetes.io/region`
+* `beta.kubernetes.io/instance-type`
+* `beta.kubernetes.io/os`
+* `beta.kubernetes.io/arch`
+
+## Alpha features: affinity and anti-affinity
+
+`nodeSelector` provides a very simple way to constrain pods to nodes with particular labels. The affinity/anti-affinity
+feature, currently in alpha, greatly expands the types of constraints you can express. The key enhancements are
+
+1. the language is more expressive (not just "AND of exact match")
+1. you can indicate that the rule is "soft"/"preference" rather than a hard requirement, so if the scheduler
```

Comment:
Should we remove the lines above that also reference affinity in favor of a pointer to this section?